third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -499,3 +507,36 @@ func resourceDataflowJobIsVirtualUpdate(d *schema.ResourceData) bool {
 
 	return false
 }
+
+func waitForDataflowJobToBeUpdated(d *schema.ResourceData, config *Config, replacementJobID string, timeout time.Duration) error {
+	return resource.Retry(timeout, func() *resource.RetryError {
+		project, err := getProject(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		region, err := getRegion(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		replacementJob, err := resourceDataflowJobGetJob(config, project, region, replacementJobID)
+		if err != nil {
+			if isGoogleApiErrorWithCode(err, 404) {
+				return resource.RetryableError(fmt.Errorf("the replacement job with ID %q was not found.", replacementJobID))
+			}
+			return resource.NonRetryableError(err)
+		}
+
+		state := replacementJob.CurrentState
+		switch state {
+		case "JOB_STATE_PENDING":
+			return resource.RetryableError(fmt.Errorf("the replacement job with ID %q has not yet started and has state %q.", replacementJobID, state))
+		case "JOB_STATE_FAILED":
+			return resource.NonRetryableError(fmt.Errorf("the replacement job with ID %q failed with state %q.", replacementJobID, state))
+		default:
+			log.Printf("the replacement job with ID %q has state %q.", replacementJobID, state)
```

Comment:
```suggestion
			log.Printf("[DEBUG] replacement job with ID %q has successful terminal state %q.", replacementJobID, state)
```

(needs `[DEBUG]` or else TF won't print it)

Reply:
Done.